src/domains/wallet-app/api/wallet-service/v1/_openapi.json.tpl

```diff
@@ -1,248 +1,378 @@
+              "$ref": "#/components/schemas/WalletId"
             }
           }
-        },
-        "delete": {
-          "tags": [
-            "wallets"
-          ],
-          "summary": "Deletes a wallet by ID",
-          "description": "delete a wallet",
-          "operationId": "deleteWallet",
-          "parameters": [
-            {
-              "name": "walletId",
-              "in": "path",
-              "description": "Wallet id to delete",
-              "required": true,
-              "schema": {
-                "type": "integer",
-                "format": "int64"
+        ],
+        "responses": {
+          "200": {
+            "description": "Wallet retrieved successfully",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/WalletInfo"
+                }
               }
             }
-          ],
-          "responses": {
-            "200": {
-              "description": "successful operation",
-              "content": {
-                "application/json": {
-                  "schema": {
-                    "$ref": "#/components/schemas/WalletDeleteResponse"
-                  }
+          },
+          "400": {
+            "description": "Invalid input id",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/ProblemJson"
                 }
               }
-            },
-            "400": {
-              "description": "Invalid ID supplied"
-            },
-            "404": {
-              "description": "Wallet not found"
             }
+          },
+          "404": {
+            "description": "Wallet not found"
+          },
+          "504": {
+            "description": "Timeout serving request"
           }
         }
       }
-    },
-    "components": {
-      "schemas": {
-        "Wallet": {
-          "type": "object",
-          "properties": {
-            "paymentInstrumentId": {
-              "type": "string"
-            }
+    }
+  },
+  "components": {
+    "schemas": {
+      "WalletId": {
+        "description": "Wallet identifier",
+        "type": "string",
+        "format": "uuid"
+      },
+      "WalletType": {
+        "type": "string",
+        "description": "Wallet type enumeration",
+        "enum": [
+          "CARDS"
+        ]
+      },
+      "Service": {
+        "type": "string",
+        "description": "Enumeration of services",
+        "enum": [
+          "PAGOPA"
+        ]
+      },
+      "WalletStatus": {
+        "type": "string",
+        "description": "Enumeration of wallet statuses",
+        "enum": [
+          "INITIALIZED",
+          "CREATED",
+          "ERROR"
+        ]
+      },
+      "CardPaymentInstrumentDetails": {
+        "type": "object",
+        "description": "Card payment instrument details",
+        "properties": {
+          "walletType": {
```

Comment:
duplicate? related to the parent?

Reply:
This property needs to be here in order to be used as discriminator

Reply:
Fixed with https://github.com/pagopa/pagopa-infra/pull/903/commits/56916a1fbeeb4d69c23dca5331a4af6daa3cc855